### ISSUE TITLE
Lantern crashing in HTTP serving 

### ISSUE THREAD

**Issue by myleshorton** [getlantern/lantern#2897]
This has come up in 2.0.0+stable:

```
ERROR flashlight.client: server.go:1777 http: panic serving 127.0.0.1:62588: runtime error: invalid memory address or nil pointer dereference
goroutine 2520 [running]:
net/http.func??011()
	/go/src/net/http/server.go:1130 +0xa8
github.com/getlantern/detour.(*Conn).Close(0x121d1d40, 0x0, 0x0)
	/lantern/src/github.com/getlantern/detour/detour.go:306 +0x2ab
github.com/getlantern/flashlight/client.func??007()
	/lantern/src/github.com/getlantern/flashlight/client/handler.go:62 +0x123
sync.(*Once).Do(0x124ca490, 0x124ca4a0)
	/go/src/sync/once.go:44 +0xb9
github.com/getlantern/flashlight/client.(*Client).intercept(0x11d08480, 0x3be220, 0x124b6770, 0x124b6540)
	/lantern/src/github.com/getlantern/flashlight/client/handler.go:89 +0x522
github.com/getlantern/flashlight/client.(*Client).ServeHTTP(0x11d08480, 0x3be220, 0x124b6770, 0x124b6540)
	/lantern/src/github.com/getlantern/flashlight/client/handler.go:27 +0x116
net/http.serverHandler.ServeHTTP(0x120bc740, 0x3be220, 0x124b6770, 0x124b6540)
	/go/src/net/http/server.go:1703 +0x145
net/http.(*conn).serve(0x124324e0)
	/go/src/net/http/server.go:1204 +0x9d8
created by net/http.(*Server).Serve
	/go/src/net/http/server.go:1751 +0x2ce
```

**Comment by myleshorton** [comment]
looks like DirectAddrCh is never created!

**Comment by myleshorton** [comment]
Should be fixed in #2900
